templates/solidity-frameworks/hardhat/packages/hardhat/hardhat.config.ts.template.mjs

```diff
@@ -1,190 +1,207 @@
-import { withDefaults } from "../../../../utils.js";
+import { withDefaults, stringify } from "../../../../utils.js";
 
-const contents = ({ imports, solidityVersion, networks, compilers }) => {
+function deepMerge(target, ...sources) {
+  if (!sources.length) return target;
+  const source = sources.shift();
 
-  const massagedCompilers = compilers[0]?.[0] ? JSON.stringify(compilers[0]) : '';
-  
-  const defaultCompilers = `[
-  {
-    version: "${solidityVersion[0]}",
-    settings: {
-      optimizer: {
-        enabled: true,
-        // https://docs.soliditylang.org/en/latest/using-the-compiler.html#optimizer-options
-        runs: 200,
-      },
-    },
-  },
-]`;
+  if (source && typeof source === 'object') {
+    for (const key in source) {
+      if (source[key] && typeof source[key] === 'object') {
+        if (!target[key]) {
+          target[key] = Array.isArray(source[key]) ? [] : {};
+        }
+        deepMerge(target[key], source[key]);
+      } else {
+        target[key] = source[key];
+      }
+    }
+  }
```

Comment:
Maybe we should move this to `utils.js`, 

Shall we use some library? Example: https://github.com/fastify/deepmerge ?

Reply:
Thanks for the suggestion! See [1762e6b](https://github.com/scaffold-eth/create-eth/pull/175/commits/1762e6ba738006505ae3cedfdbdf534ea4a130df)